src/utils/queue.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const debug = require('debug')('fastboot:ember-app');
+
+class Queue {
+  constructor(maxSize = 1) {
+    this.items = [];
+    this.maxSize = maxSize;
+  }
+
+  enqueue(item) {
+    if (this.items.length < this.maxSize) {
+      this.items.push(item);
+    } else {
+      debug('Ignoring adding appInstance to queue as Queue is already full!');
+    }
+  }
+
+  dequeue() {
+    if (!this.isEmpty()) {
+      return this.items.shift();
+    }
+
+    return null;
+  }
+
+  isEmpty() {
+    return this.items.length === 0;
+  }
+
+  size() {
+    return this.items.length;
+  }
+
+  isFull() {
+    return this.items.length === this.maxSize;
+  }
```

Comment:
Opportunity to leverage the Queues `this.size()` method for all the `this.items.length` checks. It looks like this method is only used for testing, but leveraging it in the actual code makes things a tad more DRY.

Reply:
Good point, let me update this. Thanks for catching this.